File: src/main/go/array/sort_color.go
package array

func sortColors(nums []int) {
	count := []int{0, 0, 0}

	for _, num := range nums {
		count[num]++
	}

	for i := len(nums) - 1; i >= 0; i-- {
		if count[2] != 0 {
			nums[i] = 2
			count[2]--
		} else if count[1] != 0 {
			nums[i] = 1
			count[1]--
		} else {
			nums[i] = 0
		}
	}
}

File: src/main/java/leetcode/array/SortColor.java
package leetcode.array;

public class SortColor {
  public void sortColors(int[] nums) {
    int[] count = new int[3];
    for (int num : nums) {
      count[num]++;
    }

    int right = nums.length - 1;
    while (right >= 0) {
      if (count[2] != 0) {
        nums[right--] = 2;
        count[2]--;
      } else if (count[1] != 0) {
        nums[right--] = 1;
        count[1]--;
      } else {
        nums[right--] = 0;
      }
    }
  }
}
